tests/migrations/test_executor.py

```diff
@@ -100,6 +100,43 @@ def test_run_with_squashed(self):
         self.assertTableNotExists("migrations_author")
         self.assertTableNotExists("migrations_book")
 
+    @override_settings(MIGRATION_MODULES={"migrations": "migrations.test_migrations_non_atomic"})
+    def test_non_atomic_migration(self):
+        """
+        Tests applying a non-atomic migration.
+        """
+        executor = MigrationExecutor(connection)
+        with self.assertRaises(Exception):
```

Comment:
Is there a reason you didn't use `AbortMigrationError` here? Adding a message to that exception and using `assertRaisesMessage` could make it easier to track the test to the migration. I'm not sure the custom class adds much ... probably something like `RuntimeError` (with a message) would be just as good.